Project-1/Peihong_Xie_Project_1_50375781.md

```diff
@@ -0,0 +1,249 @@
+| 1  | 1 | 1 | 1 | 1 | 1 | 1 | 1 |
+| 1  | 1 | 1 | 0 | 1 | 1 | 0 | 1 |
+| 1  | 1 | 0 | 1 | 1 | 0 | 1 | 1 |
+| 1  | 0 | 1 | 1 | 0 | 0 | 1 | 1 |
+| 0  | 1 | 1 | 1 | 1 | 1 | 0 | 1 |
+| 1  | 1 | 0 | 0 | 1 | 0 | 0 | 0 |
+| 1  | 0 | 1 | 0 | 0 | 0 | 0 | 0 |
+| 0  | 0 | 1 | 1 | 1 | 1 | 0 | 1 |
+| 0  | 1 | 1 | 0 | 1 | 1 | 0 | 1 |
+| 0  | 1 | 0 | 1 | 1 | 0 | 0 | 0 |
+| 1  | 0 | 0 | 1 | 0 | 0 | 1 | 1 |
+| 0  | 0 | 0 | 1 | 1 | 0 | 0 | 0 |
+| 0  | 0 | 1 | 0 | 1 | 1 | 0 | 1 |
+| 0  | 1 | 0 | 0 | 1 | 0 | 0 | 0 |
+| 1  | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
+| 0  | 0 | 0 | 0 | 1 | 0 | 0 | 0 |
+
+[2] A _literal_ is an atomic formula or the negation of an atomic formula. We say a formula is in _conjunctive normal form_ (CNF) if it is the conjunction of the disjunction of literals. Find propositional logic formulas in CNF equivalent to each of the following:
+  ```(a) (A→B)→C
+  (b) (A→(B∨C))∨(C→¬A)
+  (c) (¬A∧¬B∧C)∨(¬A∧¬C)∨(B∧C)∨A 
+```
+Answer:
+
+(a) CNF: (A∨C)∧(¬B∨C)
+
+(b) CNF: B∨¬B (the CNF of a tautology is any formula of the form "p∨¬p")
+
+(c) CNF: B∨¬B (the CNF of a tautology is any formula of the form "p∨¬p")
+
+[3] Let V be the vocabulary of first-order logic consisting of a binary relation P and a unary relation F. Interpret P(x,y) as “x is a parent of y” and F(x) as “x is female.” Where possible define the following formulas in this vocabulary; where not possible, explain why: 
+  ```(a)  B(x,y) that says that x is a brother of y
+  (b)  A(x,y) that says that x is an aunt of y
+  (c)  C(x,y) that says that x and y are cousins 
+  (d)  O(x) that says that x is an only child
+  (e)  T(x) that says that x has exactly two brothers 
+```
+
+Answer:
+
+(a) B(x,y)=def. ∃u(P(u,x)∧P(u,y))∧¬F(x)∧¬(x=y)
+
+(b) A(x,y)=def. ∃u∃w(P(u,y)∧P(w,x)∧P(w,u)∧¬(x=u))∧F(x)
+
+(c) C(x,y)=def. ∃u∃w∃z(P(w,z)∧P(w,u)∧P(z,x)∧P(u,y)∧¬(z=u))∧¬(x=y)
+
+(d) O(x)=def. ∃u(P(u,x)∧∀y(P(u,y)→x=y))
+
+(e) T(x)=def. ∃u∃y∃z(¬(y=z)∧¬(x=y)∧¬(x=z)∧P(u,x)∧P(u,y)∧P(u,z)∧¬F(y)∧¬F(z))∧∀w(P(u,w)∧¬F(w)→ w=x ∨ w=y ∨ w=z)
+
+[4] Let V be a vocabulary of the attribute (concept) language with complements (ALC) consisting of a role name "parent_of" and a concept name "Male". Interpret parent_of as "x is a parent of y" and M as "x is male". Where possible define the following formulas in this vocabulary; where not possible, explain why: 
+  ```(a)  B that says that x is a brother of y
+  (b)  A that says that x is an aunt of y
+  (c)  C that says that x and y are cousins 
+  (d)  O that says that x is an only child
+  (e)  T that says that x has exactly two brothers 
+```
+
+Answer:
+
+(a) $B ≡ M\sqcap ∃parentOf^-.(∃parentOf\ge2)$
+
+(b) $A ≡ ¬M\sqcap ∃parentOf^-.((∃parentOf.(∃parentOf.M\sqcup ¬M))\sqcap∃parentOf\ge2)$
+
+(c) $C ≡ ∃parentOf^-.(∃parentOf^-.(\ge2parentOf.(parentOf.M\sqcup ¬M)))$
+
+(d) $O ≡ ∃parentOf^-.(∃parentOf\le1)$
+
+(e) $T ≡ (M\sqcap ∃parentOf^-.(\le3∃parentOf.M \sqcap \ge3∃parentOf.M))\sqcup(¬M\sqcap∃parentOf^-.(\le2∃parentOf.M \sqcap \ge2∃parentOf.M))$
+
```

Comment:
The definitions look like the best thing I got so far on description logic, but could you point out where you found the syntactical rules for applying symbols like "≥2" or the symbols to mean the reverse of a concept? I didn't seem to find them in the parts of the book I was reading

Reply:
> The definitions look like the best thing I got so far on description logic, but could you point out where you found the syntactical rules for applying symbols like "≥2" or the symbols to mean the reverse of a concept? I didn't seem to find them in the parts of the book I was reading

@giacomodecolle My line of thought comes from Baader - An Introduction to Description Logic (one of our reading materials). Please see its Section 2.5. So this means that in order to do this question, we have to read some parts which are beyond our first-week reading.

Reply:
Hi Karl, I was working on this with Ali today and we may have a problem with building the reverse of parentof_, since that is an undefined primitive and we technically have no formal way to give a value to its reverse. Starting from this we built some intermediate concepts which we can use to build the whole thing. Perhaps we can discuss it tomorrow at the meeting, in the meantime here is some tentative notes:

  (a)  B that says that x is a brother of y

  Per (Person) ≡ M ⊔ ¬M
  p2 (Parent of 2) ≡ ∃parentof. ≥ 2 ⊓ ¬(∃parentof. = 1) (this condition is meant to avoid that the two persons are the same, but I am not entirely sure it works or whether there are better ways to do so)
  
  B ≡ M ⊓ ∃p2¯.Per

  (b)  A that says that x is an aunt of y
  
  ∃p2¯.parentof.parentof.Per

  A ≡ ¬M ⊓ (∃p2¯. (parentof. ≥ 2).(parentof.Per)) ⊔ (¬parentof. Per ⊓ ∃p2¯.parentof.parentof.Per)
  
  U(uncle) ≡ M ⊓ (∃p2¯. (parentof. ≥ 2).(parentof.Per)) ⊔ ¬parentof.Per 

  (c)  C that says that x and y are cousins

  gp2(Grandparent with at least two children which each have children) ≡ ∃parentof ≥ 2.parentof.Per

  C ≡ gp2¯.per

  But this would only describe one of the two cousins, i.e. "being a cousin of someone" or "having a cousin".
  We cannot define "being cousins" in description logic.

  (d)  O that says that x is an only child  

  O ≡ ¬ ∃p2¯.Per

  (e)  T that says that x has exactly two brothers 

  This cannot be done without a role name for brother, we only have a concept name for it.